### DISCUSSION TITLE
Visual mapping different from motion

### DISCUSSION THREAD

**Discussion by AckslD** [andymass/vim-matchup#213] [Q&A] [answered]
If I'm in for example the following python file:
```python
if True:
    pass
else:
    pass
```
on the first `pass` and I do eg `di%` I get:
```python
if True:
else:
    pass
```
but if I go into insert mode and do `i%` I get the following selection (marked with `[]`):
```python
if[ True:
    pass]
else:
    pass
```
Is this the intended behaviour? Is the difference because the `di%` is linewise rather than charwise? Can one choose between linewise and charwise?

Thanks for a great plugin!

**Comment by andymass** [comment, marked as answer]
It is intended, there is some discussion at `:help matchup-feat-linewise`.

Please see if this option (remove `d` and/or `y` as necessary) works for you:

    let g:matchup_text_obj_linewise_operators = ['d', 'y']  # set to []

**Reply by AckslD** [reply]
Thanks @andymass, makes sense!
